pkg/config/app/kuma-cp/config.go

```diff
@@ -271,28 +275,11 @@ func (c *Config) Validate() error {
 		if err := c.Multizone.Global.Validate(); err != nil {
 			return errors.Wrap(err, "Multizone Global validation failed")
 		}
-	case core.Standalone:
-		if err := c.XdsServer.Validate(); err != nil {
-			return errors.Wrap(err, "Xds Server validation failed")
-		}
-		if err := c.BootstrapServer.Validate(); err != nil {
-			return errors.Wrap(err, "Bootstrap Server validation failed")
-		}
-		if err := c.MonitoringAssignmentServer.Validate(); err != nil {
-			return errors.Wrap(err, "Monitoring Assignment Server validation failed")
-		}
-		if c.Environment != core.KubernetesEnvironment && c.Environment != core.UniversalEnvironment {
-			return errors.Errorf("Environment should be either %s or %s", core.KubernetesEnvironment, core.UniversalEnvironment)
-		}
-		if err := c.Runtime.Validate(c.Environment); err != nil {
-			return errors.Wrap(err, "Runtime validation failed")
-		}
-		if err := c.Metrics.Validate(); err != nil {
-			return errors.Wrap(err, "Metrics validation failed")
-		}
 	case core.Zone:
-		if err := c.Multizone.Zone.Validate(); err != nil {
-			return errors.Wrap(err, "Multizone Zone validation failed")
+		if c.IsFederatedZoneCP() {
```

Comment:
Isn't the validation going to not happen if only one of globalAddress or Zone name is set?

Reply:
I reworked this validation.